stp/stp_main.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 Broadcom. The term "Broadcom" refers to Broadcom Inc. and/or
+ * its subsidiaries.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "stp_main.h"
+
+STPD_CONTEXT stpd_context;
+
+int stpd_ipc_init()
+{
+    struct sockaddr_un sa;
+    int ret;
+    struct event *ipc_event = 0;
+
+    unlink(STPD_SOCK_NAME);
+    g_stpd_ipc_handle = socket(AF_UNIX, SOCK_DGRAM, 0);
+    if (!g_stpd_ipc_handle)
+    {
+        STP_LOG_ERR("ipc socket error %s", strerror(errno));
+        return -1;
+    }
+
+    // setup socket address structure
+    memset(&sa, 0, sizeof(struct sockaddr_un));
+    sa.sun_family = AF_UNIX;
+    strncpy(sa.sun_path, STPD_SOCK_NAME, sizeof(sa.sun_path) - 1);
+
+    ret = bind(g_stpd_ipc_handle, (struct sockaddr *)&sa, sizeof(struct sockaddr_un));
+    if (ret == -1)
+    {
+        STP_LOG_ERR("ipc bind error %s", strerror(errno));
+        close(g_stpd_ipc_handle);
+        return -1;
+    }
+
+    ipc_event = stpmgr_libevent_create(g_stpd_evbase, g_stpd_ipc_handle,
+            EV_READ|EV_PERSIST, stpmgr_recv_client_msg, (char *)"IPC", NULL);
+    if (!ipc_event)
+    {
+        STP_LOG_ERR("ipc_event Create failed");
+        return -1;
+    }
+
+    STP_LOG_DEBUG("ipc init done");
+
+    return 0;
+}
+
+void stpd_log_init()
+{
+    STP_LOG_INIT();
+    if (fopen("/stpd_dbg_reload", "r"))
+    {
+        STP_LOG_SET_LEVEL(STP_LOG_LEVEL_DEBUG);
+    }
+    else
+        STP_LOG_SET_LEVEL(STP_LOG_LEVEL_INFO);
+}
+
+int stpd_main()
```

Comment:
On Failures , in this function, any socket opened or libevent related info is not cleaned , is this fine?

Reply:
Yes, all resources will be cleaned once process is terminated